wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java

```diff
@@ -1,45 +1,198 @@
+        }
+    }
+
+    /**
+     * {@return the size of this type in memory}
+     *
+     * @throws IllegalStateException if the type cannot be stored in memory
+     */
+    public int size() {
+        switch (this) {
+            case F64:
+            case I64:
+                return 8;
+            case F32:
+            case I32:
+                return 4;
+            case V128:
+                return 16;
+            default:
+                throw new IllegalStateException("Type does not have size");
+        }
+    }
+
+    /**
+     * {@return <code>true</code> if the type is a numeric type, or <code>false</code> otherwise}
+     */
+    public boolean isNumeric() {
+        switch (this) {
+            case F64:
+            case F32:
+            case I64:
+            case I32:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * {@return <code>true</code> if the type is an integer type, or <code>false</code> otherwise}
+     */
+    public boolean isInteger() {
+        switch (this) {
+            case I64:
+            case I32:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * {@return <code>true</code> if the type is a floating-point type, or <code>false</code> otherwise}
+     */
+    public boolean isFloatingPoint() {
+        switch (this) {
+            case F64:
+            case F32:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * {@return <code>true</code> if the type is a reference type, or <code>false</code> otherwise}
+     */
+    public boolean isReference() {
+        switch (this) {
+            case FuncRef:
+            case ExternRef:
+                return true;
+            default:
+                return false;
+        }
+    }
 
-    static {
-        for (ValueType e : ValueType.values()) byId.put(e.id(), e);
+    /**
+     * {@return the <code>ValueType</code> for the given ID value, or <code>null</code> if the ID is not valid}
+     */
+    public static ValueType forId(int id) {
+        switch (id) {
+            case ID.F64:
+                return F64;
+            case ID.F32:
+                return F32;
+            case ID.I64:
+                return I64;
+            case ID.I32:
+                return I32;
+            case ID.V128:
+                return V128;
+            case ID.FuncRef:
+                return FuncRef;
+            case ID.ExternRef:
+                return ExternRef;
+            default:
+                return null;
+        }
     }
 
-    public static ValueType byId(long id) {
-        return byId.get(id);
+    /**
+     * {@return the <code>ValueType</code> for the given ID value}
+     *
+     * @throws IllegalArgumentException if the ID value does not correspond to a valid value type
+     */
+    public static ValueType forIdOrThrow(int id) {
```

Comment:
I think that this can be the only default implementation, I would call it `forId` and make the current `forId` `private` or simply remove it, wdyt?